src/moonlink_service/src/test.rs

```diff
@@ -348,6 +349,102 @@ async fn test_optimize_table_on_data_mode() {
     run_optimize_table_test("data").await;
 }
 
+/// Util function to create snapshot via REST API.
+async fn create_snapshot(client: &reqwest::Client, database: &str, table: &str, lsn: u64) {
+    let payload = get_create_snapshot_payload(database, table, lsn);
+    let crafted_src_table_name = format!("{database}.{table}");
+    let response = client
+        .post(format!(
+            "{REST_ADDR}/tables/{crafted_src_table_name}/snapshot"
+        ))
+        .header("content-type", "application/json")
+        .json(&payload)
+        .send()
+        .await
+        .unwrap();
+    assert!(
+        response.status().is_success(),
+        "Response status is {response:?}"
+    );
+}
+
+/// Test Create Snapshot
+#[tokio::test]
+#[serial]
+async fn test_create_snapshot() {
+    cleanup_directory(&get_moonlink_backend_dir()).await;
+    let config = get_service_config();
+    tokio::spawn(async move {
+        start_with_config(config).await.unwrap();
+    });
+    test_readiness_probe().await;
+
+    // Create test table.
+    let client = reqwest::Client::new();
+    create_table(&client, DATABASE, TABLE).await;
+
+    // Ingest some data.
+    let insert_payload = json!({
+        "operation": "insert",
+        "request_mode": "sync",
+        "data": {
+            "id": 1,
+            "name": "Alice Johnson",
+            "email": "alice@example.com",
+            "age": 30
+        }
+    });
+    let crafted_src_table_name = format!("{DATABASE}.{TABLE}");
+    let response = client
+        .post(format!("{REST_ADDR}/ingest/{crafted_src_table_name}"))
+        .header("content-type", "application/json")
+        .json(&insert_payload)
+        .send()
+        .await
+        .unwrap();
+    assert!(
+        response.status().is_success(),
+        "Response status is {response:?}"
+    );
+    let response: IngestResponse = response.json().await.unwrap();
+    assert_eq!(response.lsn, Some(2));
```

Comment:
I'm hesitant if it's good idea to validate LSN, since it's completely server side implementation detail and it's subject to change in the future.